Applications/Spire/Source/Ui/EditableTableView.cpp

```diff
@@ -0,0 +1,334 @@
+        (is_a<TextBox>() && !(+Any() << is_a<ListItem>()))).
+      set(horizontal_padding(scale_width(8)));
+    style.get(Any() > is_a<EditableBox>()).
+      set(horizontal_padding(scale_width(8)));
+    style.get(Any() > DeleteButton()).
+      set(Visibility(Visibility::INVISIBLE));
+    style.get(Any() > DeleteButton() > is_a<Box>()).
+      set(BackgroundColor(QColor(Qt::transparent))).
+      set(horizontal_padding(scale_width(2))).
+      set(vertical_padding(scale_height(2)));
+    style.get(Any() > DeleteButton() > is_a<Icon>()).
+      set(BackgroundColor(QColor(Qt::transparent)));
+    style.get((Any() > DeleteButton()) << is_a<TableItem>()).
+      set(padding(scale_width(2)));
+    style.get(Any() > (Row() && Hover())).
+      set(BackgroundColor(0xF2F2FF));
+    style.get(Any() > (Row() && Hover()) > DeleteButton()).
+      set(Visibility(Visibility::VISIBLE));
+    style.get((Any() > (Row() && Hover())) > DeleteButton() > is_a<Icon>()).
+      set(Fill(QColor(0x535353)));
+    style.get(Any() > (Row() && Hover()) > DeleteButton() >
+        (is_a<Icon>() && Hover())).
+      set(BackgroundColor(QColor(0xDFDFEB))).
+      set(Fill(QColor(0xB71C1C)));
+    style.get(Any() > Current()).
+      set(BackgroundColor(Qt::transparent));
+    style.get(Any() > HoverItem()).set(border_color(QColor(0xA0A0A0)));
+    style.get((Any() > DeleteButton()) << (HoverItem() || Current())).
+      set(border_color(QColor(Qt::transparent)));
+    style.get((Any() > EmptyCell()) << (HoverItem() || Current())).
+      set(border_color(QColor(Qt::transparent)));
+    style.get(Any() > CurrentRow() > DeleteButton()).
+      set(Visibility(Visibility::VISIBLE));
+    style.get(Any() > CurrentRow() > DeleteButton() > is_a<Icon>()).
+      set(Fill(QColor(0x535353)));
+    style.get(Any() > CurrentRow() > DeleteButton() >
+        (is_a<Icon>() && Hover())).
+      set(BackgroundColor(QColor(0xD0CEEB))).
+      set(Fill(QColor(0xB71C1C)));
+    style.get(Any() > CurrentRow()).set(BackgroundColor(QColor(0xE2E0FF)));
+    style.get(Any() > CurrentColumn()).set(BackgroundColor(Qt::transparent));
+    return style;
+  }
+
+  QWidget* make_empty_cell() {
+    auto cell = new QWidget();
+    cell->setSizePolicy(QSizePolicy::Expanding, QSizePolicy::Expanding);
+    match(*cell, EmptyCell());
+    return cell;
+  }
+}
+
+struct EditableTableModel : TableModel {
+  std::shared_ptr<TableModel> m_source;
+  std::shared_ptr<EditableTableView::HeaderModel> m_header;
+  TableModelTransactionLog m_transaction;
+  scoped_connection m_source_connection;
+
+  explicit EditableTableModel(std::shared_ptr<TableModel> source,
+    std::shared_ptr<EditableTableView::HeaderModel> header)
+    : m_source(std::move(source)),
+      m_header(std::move(header)),
+      m_source_connection(m_source->connect_operation_signal(
+        std::bind_front(&EditableTableModel::on_operation, this))) {}
+
+  int get_row_size() const {
+    return m_source->get_row_size() + 1;
```

Comment:
Is this row added in order to be able to add a row? If so remove it since it doesn't involve deletion.